package/MDAnalysis/topology/TOPParser.py

```diff
@@ -453,7 +453,12 @@ def parse_charges(self, num_per_record, numlines):
         vals = self.parsesection_mapper(numlines, lambda x: float(x))
         charges = np.array(vals, dtype=np.float32)
         charges /= 18.2223  # to electron charge units
-        attr = Charges(charges)
+        # in practice, the precision of partial charges
+        # in MD simulation forcefields is typically limited
+        # to a few decimal places (i.e., 1e-4 is already optimistic)
```

Comment:
needs a citation perhaps, but only if there's an appetite for moving forward

Reply:
Our recent work shows that changes on the order of 1e-3 can have large impacts on the free energy. I've seen (unreported) cases where 5e-4 does subtle changes, probably 1e-5 is the best bet?

Probably @lilyminium is the best person to weigh in on this though.

Reply:
Just some quick thoughts below, apologies I've just skimmed this issue/PR and I'm in the middle of travel. 

For the same reasons Oliver and Irfan have mentioned I'm not super keen on auto-rounding, especially to something as low as 4 dp -- just to throw out an example, OpenFF's FFs will assign charges with more dp than that and can be written out to AMBER formats (and we ensure integer charge at the molecule level but not necessarily residue level). Could rounding be a kwarg that's passed to the TOPParser instead? Then I'd be 100% in favour of having this capability in the code.

Reply:
I see, thanks for the clarification Oliver -- I was just looking at the changed lines in the PR where the kwarg is passed into the Charge attr on top-parsing. So as I understand it atom charges aren't rounded in this PR, but the residue-rounding is set on parsing. I agree that's not super intuitive and that making the user just round charges themselves is probably easier and clearer.